src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loaded, setLoaded] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const result = await getTodos();
+
+      const filteredResult = result
+        .filter(
+          todo =>
+            selectedOption === 'all' ||
+            (selectedOption === 'active' ? !todo.completed : todo.completed),
+        )
+        .filter(
+          todo =>
+            !query ||
+            todo.title
+              .trim()
+              .toLowerCase()
+              .includes(query.trim().toLowerCase()),
+        );
+
+      setTodos(filteredResult);
+      setLoaded(true);
+    };
+
+    fetchTodos();
+  }, [selectedOption, query]);
```

Comment:
The dependency array for `useEffect` includes `selectedOption` and `query`, which is correct for filtering. Ensure that any other state or props that affect the fetching logic are also included here if needed.